### PR TITLE
Cyborg Module Tool List Reorganization

### PR DESCRIPTION
**Pull request by ImogenOC**
## **About the PR**

Module tool lists load in two steps: common tools and job tools.
the common tools are the tools shown on the 'blank' module when looked at in module rewriter.
These currently load at the top of all borg modules, and then the individual lists for each module type load below it.

This PR includes changes to the common tool list, changing the common tool loadout to
Typewriter
Cell Cables
Atmospheric Analyzer
Reagent Scanner
Flash Light
Health Analyzer
Omnitool.

As well as shifting around the tools in the job modules themselves.
All job modules are affected. 

Two modules, mining and medical, have had their common tools inheritance revoked for the purpose of better organization.

## **Why's this needed?**

image

tl;dr I don't like scrolling and this system would be better as a gridview, not a scroll wheel, but thats outside of my capability right now.

The default loadout results in a handful of situations where more menu scrolling is necessary to handle a given situation than should really be necessary. While there has been talk of a cloud save system for module loadouts, it is not currently implemented or underway. This intends to stop gap that and act as a baseline for future module layouts for players.

Using my own experiences with hundreds of hours of cyborg shell game play, I've adjusted the list layouts.
This includes the decision to remove the common tool inheritance from Medical and Engineering, and perhaps others, to account for common borg responsibilities that the player encounters. 

**I WOULD APPRECIATE ADDITIONAL INPUT ON THE SPECIFIC LISTINGS.**

There is a methodology to the reorganization; concentrating the most commonly used tools together in groups of at least 7, biasing the most commonly used tools towards the middle of the list, and then shifting respective sub-jobs up and down in the list by order of engagement from my own experience. I'm hoping to encourage the player's attention to naturally keep the scroll closer to the middle in an effort to remove tedious two+ full scrolls on modules like brobo, engineering, and civilian

Tools have been sorted to decrease scroll distance in most cases that I can think of. If you are looking for one tool, tools with a relevant activity should be as close as possible.

image

Seeing as how this may also result in some borg item TODO's getting resolved as well:
Resolved TODO relating to a 'more versatile pen' comment under civilian module
## **Requests**
Input on further changes to the layout
Assistance in setting a default view position for each module.
Scroll through / carousel scrolling, not a distinct top / bottom of list would be a delight.

## **Changelog** 

```changelog
(u)Moonridden
(*)Changed loadout orders for all public cyborg modules in an effort to decrease overall scrolling
```

 [SILICON] [UI] [Code-Quality] [Input-Wanted]

### CODE REVIEW

**Comment by 444Portal** [comment]
I've already said my thoughts in imcoder but just to recap for anyone looking at the PR outside of this discussion: I don't think this does much other than create confusion on why the default has suddenly changed. Scrolling isn't good but I don't think just rearranging things so you scroll slightly less is an ideal fix. There was talk of Module Rewriter Cloud Saves which would be the middle ground by just allowing both to be done easily.

File: code/modules/robotics/robot/module_tool_creator/modules.dm [hunk 1]
```diff
@@ -9,75 +9,74 @@
...
-		/obj/item/instrument/whistle,
-		/obj/item/item_box/assorted/stickers/robot,
 		// TODO: security grenade fabricator?!
 		// /obj/item/handcuffs/tape_roll/crappy,
-		/obj/item/c_tube, // TODO: make slightly buffed version?
-		/obj/item/gun/kinetic/foamdartgun/borg,
-		/obj/item/device/detective_scanner,
-		/obj/item/device/audio_log, // TODO: make larger and non-ejectable version?
-		/obj/item/device/camera_viewer/security,
-		/obj/item/hand_labeler,
```
Comment (Vincent983) [inline code/modules/robotics/robot/module_tool_creator/modules.dm:35]: You forgot to readd this.

File: code/modules/robotics/robot/module_tool_creator/modules.dm [hunk 2]
```diff
@@ -86,110 +85,131 @@
...
 		/obj/item/device/light/flashlight,
-		/obj/item/tool/omnitool/silicon,
 		/obj/item/device/analyzer/healthanalyzer/upgraded,
-		/obj/item/device/reagentscanner,
-		/obj/item/device/analyzer/atmospheric/upgraded,
-		/obj/item/robojumper,
-		/obj/item/portable_typewriter/borg,
-	)
-
-/datum/robot/module_tool_creator/recursive/module/construction_ai
```
Comment (Vincent983) [inline code/modules/robotics/robot/module_tool_creator/modules.dm:102]: Don't remove the construction modules, one spawns in an oshan/nadir prefab.

File: code/modules/robotics/robot/module_tool_creator/modules.dm [hunk 3]
```diff
@@ -86,110 +85,131 @@
...
 
 // engineer. mechanic.
+// DOES NOT inherit common tools
 /datum/robot/module_tool_creator/recursive/module/engineering
 	definitions = list(
-		/obj/item/atmosporter,
+		/obj/item/portable_typewriter/borg,
+		/obj/item/device/analyzer/healthanalyzer/upgraded,
+		/obj/item/device/analyzer/atmospheric/upgraded,
+		/obj/item/device/reagentscanner,
```
Comment (Vincent983) [inline code/modules/robotics/robot/module_tool_creator/modules.dm:111]: I recommend moving the reagent scanner above the atmospheric analyzer, due to the analyzer being used more for making sure an area is repressurized and whatnot.

File: code/modules/robotics/robot/module_tool_creator/modules.dm (hunk 1 above)
Comment (ImogenOC) [reply code/modules/robotics/robot/module_tool_creator/modules.dm:35]: Resolved, thanks for catching.

File: code/modules/robotics/robot/module_tool_creator/modules.dm (hunk 3 above)
Comment (ImogenOC) [reply code/modules/robotics/robot/module_tool_creator/modules.dm:111]: Moved as suggested.

**Comment by 444Portal** [comment]
As I've had time to think on it more - I don't know, I don't hate it but I think it'll probably be fine for most and work better in a good couple of cases. My only issues seem to be personal gripes about how I handle organization.

**Comment by ImogenOC** [comment]
> As I've had time to think on it more - I don't know, I don't hate it but I think it'll probably be fine for most and work better in a good couple of cases. My only issues seem to be personal gripes about how I handle organization.

I'm glad you had a chance to warm up to the idea. I'm going to try a few changes that shake up even my own familiarity (again, with medical). Please do suggest any specific location changes you can think of.

**Comment by ImogenOC** [comment]
I haven't gotten anymore input, this is the only other revision that I can come up with.

Current:
image

Possible alternative / improvement.
image
Flipping the menders closer to the surgical tools improves the feeling during surgery.
This is, admittedly, a tough sell even for myself, but it is an improvement.
I'm sure we would get used to it.

**Comment by AXLFAN2000** [comment]
Civborg looks good to me (it's the only one I play)
